File: app-cpp/src/file-retrieval-client.cpp
#include <iostream>

#include "ClientProcessingEngine.hpp"
#include "ClientAppInterface.hpp"

int main(int argc, char** argv)
{
    std::shared_ptr<ClientProcessingEngine> engine = std::make_shared<ClientProcessingEngine>();
    std::shared_ptr<ClientAppInterface> interface = std::make_shared<ClientAppInterface>(engine);

    interface->readCommands();

    return 0;
}

File: app-java/src/main/java/csc435/app/FileRetrievalClient.java
package csc435.app;

public class FileRetrievalClient
{
    public static void main(String[] args)
    {
        ClientProcessingEngine engine = new ClientProcessingEngine();
        ClientAppInterface appInterface = new ClientAppInterface(engine);
        
        // read commands from the user
        appInterface.readCommands();
    }
}
